### ISSUE TITLE
AWS Bedrock provider doesn't work unless using `bedrockOptions`

### ISSUE THREAD

**Issue by elie222** [vercel/ai#3018]
### Description

This works:

```
createAmazonBedrock({
          bedrockOptions: {
            region: env.BEDROCK_REGION,
            credentials: {
              accessKeyId: env.BEDROCK_ACCESS_KEY,
              secretAccessKey: env.BEDROCK_SECRET_KEY,
            },
          },
        })(model)
```
        
This works when running locally, but not when deploying to Vercel:

```
createAmazonBedrock({
            region: env.BEDROCK_REGION,
              accessKeyId: env.BEDROCK_ACCESS_KEY,
              secretAccessKey: env.BEDROCK_SECRET_KEY,
          },
        })(model)
```

I imagine the issue is that Vercel is setting a key that causes the error (https://vercel.com/docs/projects/environment-variables/reserved-environment-variables#allowed-environment-variables). But I didn't manage to track down the error. `bedrockOptions` worked for me though.

**Comment by shaper** [comment]
Notes while investigating though I don't have a proposed fix yet:

I can reproduce the issue. As you suggest, it looks like an `AWS_SESSION_TOKEN` env var is set in the deployment but not when running locally. This is [used by AI SDK logic](https://github.com/vercel/ai/blob/8c690fe58cba7f91a54282aced267c2fd8b4200d/packages/amazon-bedrock/src/bedrock-provider.ts#L80) as the default value for `sessionToken` when instantiating the AWS `BedrockRuntimeClient`.

Both of your examples (with the noted typo below fixed) work for me locally. The second one seems to execute one query successfully but subsequent queries fail, perhaps due to re-use of a session token intended for refresh (though passing the session token along at all is unintended AFAIK and so part of the bug).

I'm not sure what's setting the env var: Vercel's containing environment, the AWS lib, or something else.

Side note -- there is an extra closing curly brace in your doesn't-work example above. It looks like a typo from removing the `credentials` sub-object. Here is the corrected second example:

```
createAmazonBedrock({
  region: env.BEDROCK_REGION,
  accessKeyId: env.BEDROCK_ACCESS_KEY,
  secretAccessKey: env.BEDROCK_SECRET_KEY,
})(model)
```

**Comment by shaper** [comment]
For an AI SDK fix, I am wondering whether perhaps we should remove the fallback to `AWS_XXX` vars in this provider, since as it is we risk clashing with values set by an external runtime that differ from what the user intends. The basic problem is that one typically has some creds one wants to use for interacting with Bedrock (say creds set A), but here we're running in a serverless function that has a bunch of env vars already set (say creds set B) and we can unintentionally partially merge the two.

It's only the token that's problematic currently, but if we special-case it (don't use its value as a fallback since it can be set outside of the user's control) but leave the others, then we have an API that could surprise a developer.

We could prefix the default env var names with `AI_SDK_` to avoid conflict, but this would be the first provider where we'd be doing that.

**Comment by elie222** [comment]
> For an AI SDK fix, I am wondering whether perhaps we should remove the fallback to `AWS_XXX` vars in this provider, since as it is we risk clashing with values set by an external runtime that differ from what the user intends. The basic problem is that one typically has some creds one wants to use for interacting with Bedrock (say creds set A), but here we're running in a serverless function that has a bunch of env vars already set (say creds set B) and we can unintentionally partially merge the two.
> 
> It's only the token that's problematic currently, but if we special-case it (don't use its value as a fallback since it can be set outside of the user's control) but leave the others, then we have an API that could surprise a developer.
> 
> We could prefix the default env var names with `AI_SDK_` to avoid conflict, but this would be the first provider where we'd be doing that.

Yes, I agree it should be renamed as it's already in use by Vercel for other purposes. And many people using this library will encounter that issue.

**Comment by Lantianyou** [comment]
Hi @lgrammel as bedrockOptions is removed in languagemodelv2, it would be real awesome if this issue can be fixed. Otherwise, would be impossible to deploy to vercel

**Comment by aayush-kapoor** [comment]
Marking this as closed for now - if you face any regression or have a similar problem after migrating to the latest AI SDK version, please feel free to open a new issue (with possible reproduction)

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This reports a concrete provider failure caused by current credential fallback behavior in Vercel deployments, rather than merely requesting a new capability. The later closure does not indicate that the original report was not a bug.

### Evidence
- The explicit Bedrock credentials configuration works locally but fails after deployment to Vercel.
- Investigation reproduced the failure and identified that the provider unintentionally combines supplied credentials with Vercel's `AWS_SESSION_TOKEN` environment variable.
- A maintainer characterized passing the fallback session token as unintended and “part of the bug.”
